.github/workflows/push.yml

```diff
@@ -17,12 +21,40 @@ jobs:
         if: ${{(github.ref == 'refs/heads/master') && !(startsWith(github.event.head_commit.message, 'release:'))}}
         run: echo "${{ secrets.GITHUB_TOKEN }}" | docker login ghcr.io -u $ --password-stdin
 
-      - name: Publish Containers
+      - name: Publish Containers to GHCR
         if: ${{(github.ref == 'refs/heads/master') && !(startsWith(github.event.head_commit.message, 'release:'))}}
         run: |
           GIT_VERSION=$(git describe --tags --always --match 'v*')
+          echo "GIT_VERSION=$GIT_VERSION" >> $GITHUB_ENV
 
           docker tag argo-tasks ghcr.io/linz/argo-tasks:latest
           docker tag argo-tasks ghcr.io/linz/argo-tasks:${GIT_VERSION}
 
           docker push --all-tags ghcr.io/linz/argo-tasks
+
+      - name: Configure AWS Credentials
+        if: ${{(github.ref == 'refs/heads/master') && !(startsWith(github.event.head_commit.message, 'release:'))}}
+        uses: aws-actions/configure-aws-credentials@v2
+        with:
+          aws-region: ap-southeast-2
+          mask-aws-account-id: true
+          role-to-assume: ${{ secrets.AWS_CI_ROLE }}
+
+      - name: Login to Amazon ECR
+        if: ${{(github.ref == 'refs/heads/master') && !(startsWith(github.event.head_commit.message, 'release:'))}}
+        id: login-ecr
+        uses: aws-actions/amazon-ecr-login@v1
+
+      - name: Publish Containers to ECR
+        if: ${{(github.ref == 'refs/heads/master') && !(startsWith(github.event.head_commit.message, 'release:'))}}
+        env:
+          REGISTRY: ${{ steps.login-ecr.outputs.registry }}
+          REPOSITORY: eks
+          IMAGE_TAG: ${{ github.sha }}
```

Comment:
what is this used for?

Reply:
An extra step was required for ECR, just configuring the aws credentials wasn't enough. 
Specifically what these env lines do I am not sure, I followed the instructions here https://github.com/aws-actions/amazon-ecr-login#examples-of-usage and changed the repository name to our repository name `eks`

Reply:
looking at the examples, the env vars are used in the following commands, but we dont use them here at all so we dont need them.

however I think this one could be useful           
```
REGISTRY: ${{ steps.login-ecr.outputs.registry }}
```

I think it could be used instead of `${{ secrets.AWS_ACCOUNT_ID_PROD }}.dkr.ecr.ap-southeast-2.amazonaws.com` I think it would resolve to the same string?

Reply:
oh I see, yes that works and look a lot better!

Reply:
you could have left the `eks`, its a fair bit shorter than  `${{ env.REPOSITORY }}` :D 

if you wanted to you could make them all env vars (including the repo name part of the tag) then its just a copy paste between the code bases and change the env vars.